notify/src/subscription/compounded.rs

```diff
@@ -144,79 +144,81 @@ impl Subscription for VirtualChainChangedSubscription {
-                        let addresses: HashSet<Address> = scope.addresses.drain(0..).collect();
-                        for address in addresses {
-                            self.addresses.entry(address.clone()).and_modify(|counter| *counter += 1).or_insert_with(|| {
-                                added.push(address);
-                                1
-                            });
-                        }
+                        let added = self.register(scope.addresses, context).expect("compounded always registers");
                         if !added.is_empty() && self.all == 0 {
-                            return Some(Mutation::new(Command::Start, Scope::UtxosChanged(UtxosChangedScope::new(added))));
+                            return Some(Mutation::new(Command::Start, UtxosChangedScope::new(added).into()));
                         }
                     }
                 }
                 Command::Stop => {
                     if !scope.addresses.is_empty() {
                         // Remove(R)
-                        let mut removed = vec![];
-                        // Make sure no duplicate exists in addresses
-                        let addresses: HashSet<Address> = scope.addresses.drain(0..).collect();
-                        for address in addresses {
-                            assert!(self.addresses.contains_key(&address));
-                            self.addresses.entry(address.clone()).and_modify(|counter| {
-                                *counter -= 1;
-                                if *counter == 0 {
-                                    removed.push(address);
-                                }
-                            });
-                        }
-                        // Cleanup self.addresses
-                        removed.iter().for_each(|x| {
-                            self.addresses.remove(x);
-                        });
+                        let removed = self.unregister(scope.addresses, context);
                         if !removed.is_empty() && self.all == 0 {
-                            return Some(Mutation::new(Command::Stop, Scope::UtxosChanged(UtxosChangedScope::new(removed))));
+                            return Some(Mutation::new(Command::Stop, UtxosChangedScope::new(removed).into()));
                         }
                     } else {
                         // Remove All
                         assert!(self.all > 0);
                         self.all -= 1;
                         if self.all == 0 {
-                            if !self.addresses.is_empty() {
-                                return Some(Mutation::new(
-                                    Command::Start,
-                                    Scope::UtxosChanged(UtxosChangedScope::new(self.addresses.keys().cloned().collect())),
-                                ));
+                            let addresses = self.to_addresses(Prefix::Mainnet, context);
```

Comment:
`Prefix::Mainnet`: the prefix should be dynamic and based on actual network (this should be fixed in multiple places)

Reply:
Fully aware of this issue.

This has no impact on the expected correct behavior since the tracker stores `ScriptPublicKey` instead of `Address` internally, stripping away the `Prefix` of the addresses.

Anyway I agree that is looks weird and even buggy (though functionally correct) and it should eventually be fixed. This though requires changes in a lot of places in the code since we do not have the current network and its matching address prefix handy all over the place and there is some internal challenge in gRPC & wRPC to rpc-core conversions.

We might postpone that for a following PR.